### DISCUSSION TITLE
Call base class function from inherited class.

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/brighterscript#872] [Q&A]
How do you call a base class function from an inherited class?  The function is not overridden.  I've tried using the following:

`super.<function name>`

but, I get the following:

`Exception has occurred.
Member function not found in BrightScript Component or interface.`

**Comment by rrirower** [comment]
I found a workaround.  I'm not sure if it's the correct way to accomplish this, but, it works.  I had to create an overridden function with the same name within the inherited class.  I call the base class using 'super' within the inherited class.  I had assumed that the inherited class had the same access to the base class functions.  By that, I mean I assumed that I could call the base class inherited function without having to define a "dummy" function in the inherited class.

**Reply by TwitchBronBron** [reply]
Yeah, this sounds like a bug in brighterscript. Do you think you'd be willing to submit a PR to fix the issue in brighterscript? You should _definitely_ be able to call a super function from the child class _without_ overriding it in the child.

**Reply by rrirower** [reply]
Are you asking me to submit some code to fix the problem?  If so, it may take some time as I am currently building a "wizard tool" for channel developers.

**Comment by TwitchBronBron** [comment]
Yeah that's what I was asking. I'm off work for a few weeks because we just had a baby, and I have a ton on my plate with some higher priority community initiatives, so I don't think we'll have time to fix this issue for a while. If you get a chance to fix it, I'd be happy to review the PR and cut a release. Otherwise I think you'll have to keep using the workaround for now.

**Reply by rrirower** [reply]
Congrats on the new addition to your family.  I can't promise anything, but, if I can finish what I'm working on, I'll take a closer look at this. I've got no time frame for my current work, so, it may take a while.

**Comment by markwpearce** [comment]
This will be fixed in v0.66

**Reply by rrirower** [reply]
Thanks for handling this.

**Comment by rrirower** [comment]
Any time frame on when this will be available?
